python_modules/dagster-wandb/dagster_wandb/io_manager.py

```diff
@@ -0,0 +1,565 @@
+                    raise WandbArtifactsIOManagerError(
+                        "An object pickled with 'cloudpickle' was found in the Artifact. But the module was not found. Please, make sure it's installed."
+                    )
+                with open(f"{artifact_dir}/{CLOUDPICKLE_FILENAME}", "rb") as file:
+                    input_value = cloudpickle.load(file)
+                    return input_value
+            elif os.path.exists(f"{artifact_dir}/{JOBLIB_FILENAME}"):
+                if not has_joblib:
+                    raise WandbArtifactsIOManagerError(
+                        "An object pickled with 'joblib' was found in the Artifact. But the module was not found. Please, make sure it's installed."
+                    )
+                with open(f"{artifact_dir}/{JOBLIB_FILENAME}", "rb") as file:
+                    input_value = joblib.load(file)
+                    return input_value
+            elif os.path.exists(f"{artifact_dir}/{PICKLE_FILENAME}"):
+                with open(f"{artifact_dir}/{PICKLE_FILENAME}", "rb") as file:
+                    input_value = pickle.load(file)
+                    return input_value
+
+            artifact.verify(root=artifacts_path)
+            return artifact
+
+    def handle_output(self, context: OutputContext, obj) -> None:
+        if obj is None:
+            context.log.warning(
+                "The output value passed to W&B IO Manager is empty. Ignore if expected."
+            )
+        else:
+            try:
+                self._upload_artifact(context, obj)
+            except WandbArtifactsIOManagerError as exception:
+                raise exception
+            except Exception as exception:
+                raise WandbArtifactsIOManagerError() from exception
+
+    def load_input(self, context: InputContext):
+        try:
+            return self._download_artifact(context)
+        except WandbArtifactsIOManagerError as exception:
+            raise exception
+        except Exception as exception:
+            raise WandbArtifactsIOManagerError() from exception
+
+
+@io_manager(
+    required_resource_keys={"wandb_resource", "wandb_config"},
+    description="IO manager to read and write W&B Artifacts",
+)
+def wandb_artifacts_io_manager(context: InitResourceContext):
+    wandb_client = context.resources.wandb_resource["sdk"]
+    wandb_host = context.resources.wandb_resource["host"]
+    wandb_entity = context.resources.wandb_config["entity"]
+    wandb_project = context.resources.wandb_config["project"]
+
+    wandb_run_name = None
+    wandb_run_id = None
+    wandb_run_tags = None
+    cache_duration_in_minutes = None
+    if context.resource_config is not None:
+        wandb_run_name = context.resource_config.get("run_name")
+        wandb_run_id = context.resource_config.get("run_id")
+        wandb_run_tags = context.resource_config.get("run_tags")
+        cache_duration_in_minutes = context.resource_config.get("cache_duration_in_minutes")
```

Comment:
let's put these in `config_schema` to io manager for schematization
```
@io_manager(
  ...
  config_schema=...
)
```
example: https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py#L79-L83